tokio-util/src/context.rs

```diff
@@ -0,0 +1,38 @@
+//! This module introduces `TokioContext`, which enables a simple way of using
+//! the tokio runtime with any other executor.
+
+use pin_project_lite::pin_project;
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tokio::runtime::Handle;
+
+pin_project! {
+    /// TODO: add docs. Get some inspiration?
+    pub struct TokioContext<F> {
+        #[pin]
+        inner: F,
+        handle: Handle,
+    }
+}
+
+impl<F: Future> Future for TokioContext<F> {
+    type Output = F::Output;
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        let me = self.project();
+        let handle = me.handle;
+        let fut = me.inner;
+
+        handle.enter(|| fut.poll(cx))
+    }
+}
+
+impl<F: Future> TokioContext<F> {
+    /// Creates a new `TokioContext`. Expects a future as its first argument, and a
+    /// `tokio::runtime::Handle` as a second.
```

Comment:
This doesn't explain anything you can't see by reading the definition. Perhaps talk about where you can get a `Handle`?

Reply:
Agree.